android/festago/app/src/main/java/com/festago/festago/presentation/ui/studentsverification/StudentsVerificationViewModel.kt

```diff
@@ -0,0 +1,136 @@
+package com.festago.festago.presentation.ui.studentsverification
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.festago.festago.analytics.AnalyticsHelper
+import com.festago.festago.analytics.logNetworkFailure
+import com.festago.festago.model.StudentVerificationCode
+import com.festago.festago.model.timer.Timer
+import com.festago.festago.model.timer.TimerListener
+import com.festago.festago.repository.SchoolRepository
+import com.festago.festago.repository.StudentsVerificationRepository
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class StudentsVerificationViewModel(
+    private val schoolRepository: SchoolRepository,
+    private val studentsVerificationRepository: StudentsVerificationRepository,
+    private val analyticsHelper: AnalyticsHelper,
+) : ViewModel() {
+
+    private val _uiState =
+        MutableStateFlow<StudentVerificationUiState>(StudentVerificationUiState.Loading)
+    val uiState: StateFlow<StudentVerificationUiState> = _uiState.asStateFlow()
+
+    val studentVerificationCode = MutableStateFlow("")
+
+    private val timer: Timer = Timer()
+
+    init {
+        validateStudentVerificationCode()
+    }
+
+    private fun validateStudentVerificationCode() {
+        viewModelScope.launch {
+            studentVerificationCode.collect { code ->
+                checkValidateCodeWhenSuccess(code)
+            }
+        }
+    }
+
+    private fun checkValidateCodeWhenSuccess(code: String) {
```

Comment:
check와 Validate는 중복된다고 생각이 들고 있습니다!
추가로 WhenSuccess는 캡슐화를 헤치는 경우라고 생각합니다!
```suggestion
    private fun validateCode(code: String) {
```

그리고 어느 곳에서는 StudentVerificationCode로 사용되고 여기서는 Code만 사용되고 있는데 전체적으로 통일되게하는건 어떤가요??

개인적인 생각으로는 StudentVerificationViewModel이기 때문에 
`VerificationCode`나 `StudentCode`, 그리고 `Code`로 모두 통일해도 괜찮다고 생각합니다!

Reply:
함수의 역할을 모두 드러내야 한다고 생각했는데 캡슐화를 해치는 것일 수 있겠네요. 동의합니다.